### ISSUE TITLE
Crash when running tensorflow.python.ops.gen_nn_ops.max_pool_grad_with_argmax

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59357]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.10.0, 2.13.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
Check failure with the following input arguments.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
from tensorflow.python.ops import gen_nn_ops
try:
  try:
    with tf.device('/CPU'):
      arg_0_tensor = tf.random.uniform([2, 3, 3, 1], dtype=tf.float32)
      arg_0 = tf.identity(arg_0_tensor)
      arg_1_tensor = tf.random.uniform([2, 2, 2, 1], dtype=tf.float32)
      arg_1 = tf.identity(arg_1_tensor)
      arg_2_tensor = tf.random.uniform([2, 2, 2, 1], minval=-256, maxval=257, dtype=tf.int64)
      arg_2 = tf.identity(arg_2_tensor)
      ksize_0 = 1
      ksize_1 = 2
      ksize_2 = 2
      ksize_3 = 1
      ksize = [ksize_0,ksize_1,ksize_2,ksize_3,]
      strides_0 = 1
      strides_1 = 1
      strides_2 = 1
      strides_3 = 1
      strides = [strides_0,strides_1,strides_2,strides_3,]
      padding = "VALID"
      include_batch_in_index = False
      out = gen_nn_ops.max_pool_grad_with_argmax(arg_0,arg_1,arg_2,ksize=ksize,strides=strides,padding=padding,include_batch_in_index=include_batch_in_index,)
  except Exception as e:
    print("Error:"+str(e))
  try:
    with tf.device('/GPU:0'):
      arg_0 = tf.identity(arg_0_tensor)
      arg_0 = tf.cast(arg_0, tf.float32)
      arg_1 = tf.identity(arg_1_tensor)
      arg_1 = tf.cast(arg_1, tf.float32)
      arg_2 = tf.identity(arg_2_tensor)
      arg_2 = tf.cast(arg_2, tf.int64)
      ksize = [ksize_0,ksize_1,ksize_2,ksize_3,]
      strides = [strides_0,strides_1,strides_2,strides_3,]
      gen_nn_ops.max_pool_grad_with_argmax(arg_0,arg_1,arg_2,ksize=ksize,strides=strides,padding=padding,include_batch_in_index=include_batch_in_index,)
  except Exception as e:
    print("Error:"+str(e))
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-21 12:49:08.508138: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 12:49:08.623891: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-21 12:49:09.210798: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 12:49:09.211024: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 12:49:09.211033: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-21 12:49:09.832017: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-21 12:49:09.861818: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/
2023-01-21 12:49:09.861835: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-21 12:49:09.862084: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-21 12:49:09.880100: F tensorflow/core/kernels/maxpooling_op.cc:1065] Check failed: grad_out_index >= output_start && grad_out_index < output_end Invalid output gradient index: 146, 0, 18
Aborted

```
```
</details>

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

Apologies for the delay and I was able to replicate the issue and I observed, Given code is working fine on GPU but same code is crashing on CPU so we have to dig more into this unexpected behaviour for CPU so if you have any idea or thoughts what may be the root cause for crashing code on CPU it will be greatly help us to fix the issue

I have added screenshot where code is running fine on GPU and we'll update soon about this unexpected behaviour issue for CPU. Thank you!

![image](https://user-images.githubusercontent.com/115997457/214270263-3490ff19-1698-48b2-8cc3-f9d0b82c09ae.png)

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

Apologize for the delay and I tried to run your given code on Google colab and also on `Ubuntu 22.04` with `CPU` and `GPU` and in Google colab code is getting crashed so I have attached screenshot of error log below and for `Ubuntu 22.04` with `CPU` and `GPU` I'm getting the exact same error as you mentioned in your log output. Thank you!

Here is screenshot of error log for Google Colab Run (CPU) :

![image](https://user-images.githubusercontent.com/115997457/216040485-f12b6dca-8a21-484e-b9a5-eb4baf6b9bec.png)

Here is screenshot of code run on `Ubuntu 22.04` with `CPU` and `GPU`  and I ran code multiple times but I was getting the same error 

![image](https://user-images.githubusercontent.com/115997457/216041833-da340db2-5349-44d8-ae1d-bae4976ad271.png)

**Comment by mihaimaruseac** [comment]
This is a real assertion failure that should be fixed. PRs welcome.

Not a vulnerability for which credit would be given though, given it is only an assertion failure, so impact is minimal

**Comment by mihaimaruseac** [comment]
(sorry, closed by mistake)

**Comment by mihaimaruseac** [comment]
@learning-to-play

**Comment by dmc1778** [comment]
> @learning-to-play

This bug is not fixed on 2.13.0.

**Comment by learning-to-play** [comment]
@cantonios This issue seems to be related to TF Ops. Could you help triage?

**Comment by manojssmk** [comment]
@dmc1778 
Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The issue you are getting because the minval and maxval that you are initialising in the arg_2_tensor. The values must be between 0 to 8. Kindly find the [gist](https://colab.sandbox.google.com/gist/manojssmk/c9c4eb18498d89a6490a1ed4e737da22/untitled13.ipynb)

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
